docs/source/daml/patterns/daml/MultiplePartyAgreement.daml

```diff
@@ -11,60 +11,82 @@ import Utilities
 import DA.List
 
 -- start snippet: agreement template
-template Agreement
+template ContractPlaceholder
   with
     signatories: [Party]
   where
     signatory signatories
+    ensure
+      unique signatories
+  -- The rest of the template to be agreed to would follow here
 -- end snippet: agreement template
 
 -- start snippet: first half pending template
 template Pending
   with
-    agree: Agreement
-    signed: [Party]
-    toSign: [Party]
+    finalContract: ContractPlaceholder
+    alreadySigned: [Party]
   where
-    signatory signed
-    observer toSign
+    signatory alreadySigned
+    observer finalContract.signatories
+    ensure
+      -- Can't have duplicate signatories
+      unique alreadySigned
+
+    -- The parties who need to sign is the finalContract.signatories with alreadySigned filtered out
+    let toSign = filter (`notElem` alreadySigned) finalContract.signatories
 
     choice Sign : ContractId Pending with
         signer : Party
       controller signer
         do
+          -- Check the controller is in the toSign list, and if they are, sign the Pending contract
           assert (signer `elem` toSign)
-          create this with signed = signer :: signed, toSign = delete signer toSign
+          create this with alreadySigned = signer :: alreadySigned
     -- end snippet: first half pending template
     -- start snippet: second half pending template
-    choice Finalize : ContractId Agreement with
+    choice Finalize : ContractId ContractPlaceholder with
```

Comment:
I think it is important to show how to auto-finalize the contract, even if we show this one first as the basic manipulation of `Either` and `Optional` within `Update` is a basic DAML skill.

Reply:
I'd like to do this, but within a separate pattern.